The player stepped on the campfire and reported an error

### Stack trace

```
paste your stack trace or a paste.gg link here!
```
[11:07:35] [Server thread/ERROR]: Error whilst processing packet net.minecraft.network.protocol.game.PacketPlayInFlying$PacketPlayInPosition@2f1aeadb for TheHothotsummer[/127.0.0.1:65020]
net.minecraft.ReportedException: Colliding entity with block
	at net.minecraft.world.entity.Entity.tryCheckInsideBlocks(Entity.java:981) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.world.entity.Entity.move(Entity.java:948) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleMovePlayer(ServerGamePacketListenerImpl.java:1356) ~[app:?]
	at net.minecraft.network.protocol.game.ServerboundMovePlayerPacket.handle(ServerboundMovePlayerPacket.java:114) ~[app:?]
	at net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos.handle(ServerboundMovePlayerPacket.java:42) ~[app:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:36) ~[app:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149) ~[app:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[app:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1348) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.shouldRun(MinecraftServer.java:190) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[app:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1327) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1320) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132) ~[app:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1281) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1192) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[patched_1.17.1.jar:git-Paper-163]
	at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.IllegalStateException: Unhandled damage of EntityPlayer['TheHothotsummer'/1258053, uuid='8d43f43d-a0c6-371c-ab3d-9b0d6e9231b4', l='ServerLevel[Resource]', x=-1724.27, y=64.44, z=-1702.68, cpos=[-108, -107], tl=90672, v=true, rR=null](TheHothotsummer at -1724.2692,64.4375,-1702.682) by CraftBlock{pos=BlockPosition{x=-1724, y=64, z=-1703},type=CAMPFIRE,data=Block{minecraft:campfire}[facing=west,lit=true,signal_fire=false,waterlogged=false],fluid=net.minecraft.world.level.material.FluidTypeEmpty@66236a0a} from inFire
	at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:1013) ~[patched_1.17.1.jar:git-Paper-163]
	at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:942) ~[patched_1.17.1.jar:git-Paper-163]
	at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:1132) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.world.entity.LivingEntity.damageEntity0(LivingEntity.java:2042) ~[app:?]
	at net.minecraft.world.entity.player.Player.damageEntity0(Player.java:1013) ~[app:?]
	at net.minecraft.world.entity.LivingEntity.hurt(LivingEntity.java:1350) ~[app:?]
	at net.minecraft.world.entity.player.Player.hurt(Player.java:920) ~[app:?]
	at net.minecraft.server.level.ServerPlayer.hurt(ServerPlayer.java:998) ~[app:?]
	at net.minecraft.world.level.block.CampfireBlock.entityInside(CampfireBlock.java:97) ~[app:?]
	at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.entityInside(BlockBehaviour.java:904) ~[app:?]
	at net.minecraft.world.entity.Entity.checkInsideBlocks(Entity.java:1226) ~[patched_1.17.1.jar:git-Paper-163]
	at net.minecraft.world.entity.Entity.tryCheckInsideBlocks(Entity.java:975) ~[patched_1.17.1.jar:git-Paper-163]
	... 18 more
[11:07:35] [Server thread/INFO]: TheHothotsummer lost connection: Packet processing error

### Plugin list

[https://i.loli.net/2021/08/07/BweOQ6DKJLaIl59.png](url)

### Actions to reproduce (if known)

1.Put a campfire to world 1 (Don't near spawnpoint,make sure that the campfire cannot be loaded at the spawnpoint)
2.Teleport to world 2 and wait a few seconds
3.Teleport to world 1 and teleport to campfire (integrated with the campfire)
4.Then the server will report error and stop

### Paper version

[11:26:18 INFO]: Checking version, please wait...
[11:26:19 INFO]: This server is running Paper version git-Paper-163 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: a178fb8)
You are running the latest version
>Previous version: git-Paper-138 (MC: 1.17.1)

### Agreements

- [X] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

### Other

_No response_

--- a/patches/server/0740-call-EntityDamageByBlockEvent-for-campfires.patch
+++ b/patches/server/0740-call-EntityDamageByBlockEvent-for-campfires.patch
@@ -18,3 +18,18 @@ index 2519a0f511f0a6065459cd2fe2d9a3e68e55d222..2c334ce3fc3cf86704dee9397166119a
          }
  
          super.entityInside(state, world, pos, entity);
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index d01d4ca873f8225d3503ecb102ce2eb7bfced421..622a0f23db838b839f76722df0e5480ec927b4a2 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -1009,6 +1009,10 @@ public class CraftEventFactory {
+                 cause = DamageCause.HOT_FLOOR;
+             } else if (source == DamageSource.MAGIC) {
+                 cause = DamageCause.MAGIC;
++            // Paper start
++            } else if (source == DamageSource.IN_FIRE) {
++                cause = DamageCause.FIRE;
++            // Paper end
+             } else {
+                 throw new IllegalStateException(String.format("Unhandled damage of %s by %s from %s", entity, damager, source.msgId));
+             }

